SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUpstreamChecker.java

```diff
@@ -0,0 +1,192 @@
+      }
+      LOG.info("No extra upstreams for service {}. No load balancer request sent.", singularityRequest.getId());
+      return Optional.absent();
+    } catch (Exception e) {
+      LOG.error("Could not sync for service {}. Load balancer request {} threw error: ", singularityRequest.getId(), loadBalancerRequestId, e);
+      return Optional.of(new SingularityLoadBalancerUpdate(BaragonRequestState.UNKNOWN, loadBalancerRequestId, Optional.of(String.format("Exception %s - %s", e.getClass().getSimpleName(), e.getMessage())), start, LoadBalancerMethod.CHECK_STATE, Optional.absent()));
+    }
+  }
+
+  private boolean noPendingDeploy() {
+    return deployManager.getPendingDeploys().size() == 0;
+  }
+
+  private void syncUpstreamsForService(SingularityRequest singularityRequest) {
+    if (singularityRequest.isLoadBalanced() && noPendingDeploy()) {
+      final String singularityRequestId = singularityRequest.getId();
+      LOG.info("Starting syncing of upstreams for service: {}.", singularityRequestId);
+      final Optional<String> maybeDeployId = deployManager.getInUseDeployId(singularityRequestId);
+      if (maybeDeployId.isPresent()) {
+        final String deployId = maybeDeployId.get();
+        final Optional<SingularityDeploy> maybeDeploy = deployManager.getDeploy(singularityRequestId, deployId);
+        if (maybeDeploy.isPresent()) {
+          final Optional<SingularityLoadBalancerUpdate> maybeSyncUpstreamsUpdate = syncUpstreamsForServiceHelper(singularityRequest, maybeDeploy.get(), maybeDeploy.get().getLoadBalancerUpstreamGroup());
+          if (maybeSyncUpstreamsUpdate.isPresent()) {
+            checkSyncUpstreamsState(maybeSyncUpstreamsUpdate.get().getLoadBalancerRequestId(), singularityRequestId);
+          }
+        }
+      }
+    }
+  }
+
+  private void checkSyncUpstreamsState(LoadBalancerRequestId loadBalancerRequestId, String singularityRequestId) {
+    LOG.info("Checking load balancer request to sync upstreams for service {} using a retryer until the request state is no longer waiting.", singularityRequestId);
+    Retryer<SingularityLoadBalancerUpdate> syncingRetryer = RetryerBuilder.<SingularityLoadBalancerUpdate>newBuilder()
+        .retryIfException()
+        .withWaitStrategy(WaitStrategies.fixedWait(1, TimeUnit.SECONDS))
+        .retryIfResult(IS_WAITING_STATE)
+        .build();
+    try {
+      SingularityLoadBalancerUpdate syncUpstreamsState = syncingRetryer.call(() -> lbClient.getState(loadBalancerRequestId));
+      if (syncUpstreamsState.getLoadBalancerState() == BaragonRequestState.SUCCESS){
+        LOG.info("Syncing upstreams for singularity request {} is {}.", singularityRequestId, syncUpstreamsState.toString());
+      } else {
+        LOG.error("Syncing upstreams for singularity request {} is {}.", singularityRequestId, syncUpstreamsState.toString());
```

Comment:
Can we log out the `message` (or maybe the whole `SingularityLoadBalancerUpdate` object) here as well? That will have the specific error that caused the update to fail.